software_interrupt: Add additional testing

josephlr opened the pull request:
So I just realized we already had some of the changes in #259 merged as part of #227.

This PR adds the rest of the testing code. CC: @Freax13 

Signed-off-by: Joe Richey <joerichey@google.com>

phil-opp commented:
So we accidentally resetted the `next` branch and removed the changes from #259 from it? Perhaps we should make `next` a protected branch to avoid issues like this in the future.

josephlr commented:
> So we accidentally resetted the `next` branch and removed the changes from #259 from it? Perhaps we should make `next` a protected branch to avoid issues like this in the future. _Edit:_ Seems like the branch protection was already turned on, so I'm not sure how this happened.

I just added the setting that prevents force pushing (after I realized my mistake in moving the `next` branch). I think I didn't have the permissions setup correctly last time.

josephlr commented:
@phil-opp do you think we should have the "Require pull request reviews before merging" and "Include Administrators" settings checked for the `next` and `master` branches?

phil-opp commented:
@josephlr No worries, I just tried to understand what was going on.

> do you think we should have the "Require pull request reviews before merging" and "Include Administrators" settings checked for the `next` and `master` branches? This is the setup we have for [`getrandom`](https://github.com/rust-random/getrandom/)

Sure, we can try that if you have time for regular reviews. Not directly related, but I would also like to automate the release creation process through GitHub actions (e.g. use the GitHub Api to detect whether the current crate version is present on crates.io and do a `cargo publish` if not). This way, releasing a new version would be as easy as changing the version number in the Cargo.toml through a PR.

josephlr commented:
> Sure, we can try that if you have time for regular reviews.

Given that we're starting to use this crate more at work, I think I'll be able to commit more time here. Also, can anyone in the @rust-osdev org approve changes for this repo?

I just enabled the settings and confirmed that I can no longer merge this PR without your approval.

> Not directly related, but I would also like to automate the release creation process through GitHub actions (e.g. use the GitHub Api to detect whether the current crate version is present on crates.io and do a `cargo publish` if not). This way, releasing a new version would be as easy as changing the version number in the Cargo.toml through a PR.

I think that would be an optimal way to go about things, anything that makes the release process less painful is fine by me.

phil-opp commented:
> Given that we're starting to use this crate more at work, I think I'll be able to commit more time here.

Great to hear that! I would love to hear more about how you're using this crate. Anything you can talk about?



> I just enabled the settings and confirmed that I can no longer merge this PR without your approval.

:+1: 



> I think that would be an optimal way to go about things, anything that makes the release process less painful is fine by me.

Ok, cool. I try to look into this soon.

phil-opp reviewed (approved):
Looks good!
